get_s3.go

```diff
@@ -329,34 +296,57 @@ func (g *S3Getter) parseUrl(u *url.URL) (region, bucket, path, version string, c
 	_, hasAwsSecret := u.Query()["aws_access_key_secret"]
 	_, hasAwsToken := u.Query()["aws_access_token"]
 	if hasAwsId || hasAwsSecret || hasAwsToken {
-		creds = credentials.NewStaticCredentials(
+		credentialProvider := credentials.NewStaticCredentialsProvider(
 			u.Query().Get("aws_access_key_id"),
 			u.Query().Get("aws_access_key_secret"),
 			u.Query().Get("aws_access_token"),
 		)
+		creds = &credentialProvider
 	}
 
 	return
 }
 
-func (g *S3Getter) newS3Client(
-	region string, url *url.URL, creds *credentials.Credentials,
-) (*s3.S3, error) {
-	var sess *session.Session
-
-	if profile := url.Query().Get("aws_profile"); profile != "" {
-		var err error
-		sess, err = session.NewSessionWithOptions(session.Options{
-			Profile:           profile,
-			SharedConfigState: session.SharedConfigEnable,
-		})
+func (g *S3Getter) newS3Client(ctx context.Context, region string, url *url.URL, creds *credentials.StaticCredentialsProvider) (*s3.Client, error) {
+	var cfg aws.Config
+	var err error
+
+	// We first check if the AWS_METADATA_URL is set, if it is we use it to load the config.
+	if creds == nil && os.Getenv("AWS_METADATA_URL") != "" {
+		cfg, err = config.LoadDefaultConfig(
+			ctx,
+			config.WithRegion(region),
+			config.WithEC2IMDSEndpoint(os.Getenv("AWS_METADATA_URL")),
```

Comment:
It’s unclear to me if this is the right approach and whether it provides feature parity with the current implementation of metadata URL overriding. Something to look into.